### PR TITLE
Generate subnetwork for terraform.

### PR DESCRIPTION
**Pull request by ndmckinley**
Six notes:
- Adds a few fields to Subnetwork which were present in Terraform but
not elsewhere (because they're in beta).
- Changes terraform so that it's possible to rename a field with child
fields.  Previously, the ordering of overrides was undefined, so it was
possible to rename a field from 'foo' to 'bar' - and then when it came
time to apply an override to 'foo.baz', it would crash since there's no
field called 'foo'.  Now we apply overrides in reverse order of level of
nestedness, so child fields are overriden first.
- Forces 'id_format' to be a valid import format.
- Sets subnetwork to be non-updateable, so that ansible/puppet/chef will
destroy/recreate (and so terraform will use custom-update
appropriately).
- Adds beta generation to generate-terraform.sh.  By the
zero-one-infinity rule, the *right* thing to do is to add a 'beta
detector' and start automatically picking the version in that script...
maybe next quarter.  :)
- Removed the ability to update enable_flow_logs.  According to the
docs, this never worked - patch can only update secondaryIpRange,
role (which does not exist - I filed a bug) and
allow_subnet_cidr_routes_overlap, which is a field that cannot be set at
create-time, only updated, and which I decided to omit.  In any event,
no enable_flow_logs updates.
https://cloud.google.com/compute/docs/reference/rest/beta/subnetworks/patch

-----------------------------------------------------------------
# [all]
Updates to subnetwork docs, removes broken 'update' function.
## [terraform]
Autogenerate Subnetwork.

### CODE REVIEW

File: templates/terraform/resource.erb [hunk 1]
```diff
@@ -85,18 +85,16 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
   }
   <% end -%>
 
+	obj := make(map[string]interface{})
 <% settable_properties.each do |prop| -%>
 	<%= prop.api_name -%>Prop, err := expand<%= resource_name -%><%= titlelize_property(prop) -%>(d.Get("<%= Google::StringUtils.underscore(prop.name) -%>"), d, config)
 	if err != nil {
 		return err
+	} else if v, ok := d.GetOkExists("<%= Google::StringUtils.underscore(prop.name) -%>"); ok || !reflect.DeepEqual(v, <%= prop.api_name -%>Prop) {
```
Comment (danawillow) [inline templates/terraform/resource.erb]: Let me make sure I understand the second half of the statement. This seems to handle the case where the value was not set (therefore its value will be the zero value) and its value (the zero value) is not equal to what expand produced. So this is for if it wasn't set, but we still do stuff in expand for it? So things like region would be the example case, right? Is there another type of field that would do it? Either way, mind adding a comment (in ruby blocks is fine) explaining this case?

File: .ci/magic-modules/generate-terraform.sh [hunk 2]
```diff
@@ -29,6 +29,7 @@ bundle exec compiler -p products/redis -e terraform -o "${GOPATH}/src/github.com
 
 # Resources that were already using beta APIs before they started being autogenerated
 bundle exec compiler -v beta -p products/compute -t Address -e terraform -o "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google/"
+bundle exec compiler -v beta -p products/compute -t Subnetwork -e terraform -o "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google/"
```
Comment (danawillow) [inline .ci/magic-modules/generate-terraform.sh]: nit: -t takes a list, so you should be able to just do `-t Address,Subnetwork` (probably should verify this for sure though)

File: templates/terraform/resource.erb (hunk 1 above)
Comment (ndmckinley) [reply templates/terraform/resource.erb]: You got it, and you're 100% right.  I will add a comment as soon as I am convinced, myself, that it works.  :)

**Comment by ndmckinley** [comment]
Bummer, this introduces an issue with SslProxy that looks related - I'm on it.

**Comment by ndmckinley** [comment]
Okay, to fix that I _did_ have to introduce the concept analogous to `SendEmptyValue`, which I don't _love_, but which I really do think is the only way.  Here it is.  Tests pass with this code.

**Pull request merged**